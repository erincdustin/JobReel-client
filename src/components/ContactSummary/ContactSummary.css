.saved-contact-list {
    position: relative;
    grid-column-start: 4;
    grid-column-end: 9;
    height: 100%;
    max-height: 450px;
    border: 1px solid black;
    border-radius: 10px;
    position: relative;
    background: #3C3230;
    color: #eeebd9;
    text-align: center;
    overflow-y: auto;
}

.saved-contact-controls {
    width: 50%;
    margin-left: 50%;
    margin-top: 0.5em; 
}

@media screen and (max-width: 960px) {
    .saved-contact-list {
        grid-column-start: 1;
        grid-column-end: 9;
    }
}