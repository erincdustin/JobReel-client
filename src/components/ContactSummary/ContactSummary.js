import React from 'react';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import { Label, Input } from '../Form/Form';
import Contact from '../Contact/Contact';
import './ContactSummary.css';

class ContactSummary extends React.Component {

  static contextType = JobReelContext;

  state = {
    filter: '',
    search: ''
  }

  handleStatusFilter = e => {
    if(e.target.value === 'false'){
      this.setState({ filter: false })
    } else if(e.target.value === 'true'){
      this.setState({ filter: true })
    } else {
      this.setState({ filter: '' })
    }
  }

  handleChangeSearchTerm = e => {
    this.setState({ search: e.target.value })
  }

  renderContacts = () => {
    let search = this.state.search
    let filter = this.state.filter
    let contacts = Array.from(this.context.contacts)
    if(search !== ''){
      contacts = contacts.filter(contact => contact.contact_name.toLowerCase().includes(search.toLowerCase()) || 
      contact.job_title.toLowerCase().includes(search.toLowerCase()) || 
      contact.company.toLowerCase().includes(search.toLowerCase())
    )}
    if(filter !== ''){
      contacts = contacts.filter(contact => contact.connected === filter)
    }
    contacts = contacts.map(contact => <Contact key={contact.contact_id} id={contact.contact_id} name={contact.contact_name} job_title={contact.job_title} company={contact.company} email={contact.email} linkedin={contact.linkedin} comments={contact.comments} date={contact.date_added} connected={contact.connected} date_connected={contact.date_connected} user={contact.user_id} notification={contact.notification}/>)
    
    return (
      <div>
        {contacts}
      </div>
    )
  }

  render(){
    return(
      <div className="saved-contact-list">
        <div className='savedContactFilterControls'>
          <Label id='savedContactFilterTitle'>Filter by Job Status:</Label>
          <select
            id='status-input'
            name='status'
            onChange={this.handleStatusFilter}
          >
            <option value="">N/A</option>
            <option value="false">Not Connected</option>
            <option value="true">Connected</option>
          </select>
          <br/>
          <Label id='savedContactFilterSearch'>Search:</Label>
          <Input
            type='text'
            name='saved-contact-search'
            id='saved-contact-search'
            value={this.state.search}
            onChange={this.handleChangeSearchTerm}
          />
          <br/>
          <Button onClick={() => this.context.setManualContactAdd(true)} type="button">Add Contact</Button>
        </div>
        {this.renderContacts()}
      </div>
    )
  }
}

export default ContactSummary