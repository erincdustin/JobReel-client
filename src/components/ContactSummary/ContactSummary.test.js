import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PropTypes from 'prop-types'
import ContactSummary from './ContactSummary'

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      contacts: [1, 2, 3]
    }
  })

  render(){
    return this.props.children
  }
}
describe('ContactSummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<ContactSummary />, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})

describe('Company snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><ContactSummary /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})