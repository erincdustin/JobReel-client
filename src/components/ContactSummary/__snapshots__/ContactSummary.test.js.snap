// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company snapshot test renders the UI as expected 1`] = `<ContactSummary />`;
